Core/include/Acts/Seeding/CandidatesForMiddleSp.hpp

```diff
@@ -0,0 +1,187 @@
+  /// selection criteria
+  /// @param SpB Bottom space point
+  /// @param SpM Medium space point
+  /// @param SpT Top space point
+  /// @param weight The quality of the triplet candidate
+  /// @param zOrigin The z-coordinate of the origin
+  /// @param isQuality Whether the triplet candidate is high or low quality
+  void push(external_space_point_t& SpB, external_space_point_t& SpM,
+            external_space_point_t& SpT, float weight, float zOrigin,
+            bool isQuality);
+
+  /// @brief Clear the internal storage
+  void clear();
+
+  /// @brief A function for sorting the triplet candidates from higher to lower quality
+  /// @param i1 First triplet candidate
+  /// @param i2 Second triplet candidate
+  /// @returns The comparison result
+  static bool greaterSort(const value_type& i1, const value_type& i2);
+
+ private:
+  /// @brief dding a new triplet candidate to the collection, should it satisfy the
+  /// selection criteria
+  /// @param storage The collection into which the candidate should be stored
+  /// @param n The current number of stored elements in the container
+  /// @param n_max The maximum number of elements that can be stored in the container
+  /// @param SpB The bottom space point
+  /// @param SpM The middle space point
+  /// @param SpT The top space point
+  /// @param weight The quality of the triplet candidate
+  /// @param zOrigin The z-coordinate of the origin
+  /// @param isQuality Whether the triplet candidate is high or low quality
+  void push(std::vector<std::size_t>& storage, std::size_t& n,
+            const std::size_t& n_max, external_space_point_t& SpB,
+            external_space_point_t& SpM, external_space_point_t& SpT,
+            float weight, float zOrigin, bool isQuality);
+
+  /// @brief Check if an element exists in the collection. The element to be checked
+  /// is supposed to be in the n position of the collection.
+  /// @param n Index of the requested element
+  /// @param max_size Number of elements currently stored in the collection
+  /// @returns Whether the element exists
+  bool exists(const std::size_t& n, const std::size_t& max_size) const;
+
+  /// @brief Pop an element from a collection. The removal of the element from the collection
+  /// does not imply its destruction. In fact, the number of stored elements is
+  /// simply diminished by 1. The popped element is tecnically still available
+  /// at the end of the collection.
+  /// @param storage The collection
+  /// @param current_size The current number of element stored in the collection. The function will
+  /// diminish this value by 1
+  void pop(std::vector<std::size_t>& storage, std::size_t& current_size);
+
+  /// @brief Return the weight for a candidate
+  /// @param storage The collection in which the element is stored
+  /// @param n Index of the element in the collection
+  /// @returns The weight of the candidate
+  float weight(const std::vector<std::size_t>& storage, std::size_t n) const;
```

Comment:
It is not clear why the storage needs to be passed. Isn't it available as a member of this class?

Reply:
the `storage` here is one of the two indices collections and `n` refers to the number of elements in that specific index collection. Since we have two different indices collections I preferred to pass them as inputs to the function.